sdk/eventhub/Azure.Messaging.EventHubs/tests/Primitives/EventProcessorTests.cs

```diff
@@ -22,6 +24,947 @@ namespace Azure.Messaging.EventHubs.Tests
+        public void NamespaceConstructorSetsTheIdentifier()
+        {
+            var options = new EventProcessorOptions
+            {
+                Identifier = Guid.NewGuid().ToString()
+            };
+
+            var eventProcessor = new BasicProcessorMock(65, "consumerGroup", "namespace", "hub", Mock.Of<TokenCredential>(), options);
+
+            Assert.That(eventProcessor.Identifier, Is.Not.Null);
+            Assert.That(eventProcessor.Identifier, Is.EqualTo(options.Identifier));
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventProcessor{TPartition}" />
+        ///   constructor.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(null)]
+        [TestCase("")]
+        public void ConnectionStringConstructorCreatesTheIdentifierWhenNotSpecified(string identifier)
+        {
+            var options = new EventProcessorOptions
+            {
+                Identifier = identifier
+            };
+
+            var eventProcessor = new BasicProcessorMock(34, "consumerGroup", "Endpoint=sb://somehost.com;SharedAccessKeyName=ABC;SharedAccessKey=123;EntityPath=somehub", options);
+
+            Assert.That(eventProcessor.Identifier, Is.Not.Null);
+            Assert.That(eventProcessor.Identifier, Is.Not.Empty);
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventProcessor{TPartition}" />
+        ///   constructor.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(null)]
+        [TestCase("")]
+        public void NamespaceConstructorCreatesTheIdentifierWhenNotSpecified(string identifier)
+        {
+            var options = new EventProcessorOptions
+            {
+                Identifier = identifier
+            };
+
+            var eventProcessor = new BasicProcessorMock(665, "consumerGroup", "namespace", "hub", Mock.Of<TokenCredential>(), options);
+
+            Assert.That(eventProcessor.Identifier, Is.Not.Null);
+            Assert.That(eventProcessor.Identifier, Is.Not.Empty);
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventProcessor{TPartition}" />
+        ///   constructor.
+        /// </summary>
+        ///
+        [Test]
+        public void ConnectionStringConstructorCopiesTheIdentifier()
+        {
+            var options = new EventProcessorOptions { Identifier = Guid.NewGuid().ToString() };
+            var eventProcessor = new BasicProcessorMock(33, "consumerGroup", "Endpoint=sb://somehost.com;SharedAccessKeyName=ABC;SharedAccessKey=123;EntityPath=somehub", options);
+
+            Assert.That(eventProcessor.Identifier, Is.EqualTo(options.Identifier));
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventProcessor{TPartition}" />
+        ///   constructor.
+        /// </summary>
+        ///
+        [Test]
+        public void NamespaceConstructorCopiesTheIdentifier()
```

Comment:
These tests are the same as `...SetsTheIdentifier` ones, aren't they?

Reply:
Hrmm.... looks like an assert was overlooked.